src/Microsoft.AspNet.Mvc.Core/DefaultControllerActionArgumentBinder.cs

```diff
@@ -118,19 +117,18 @@ private static void CallPropertyAddRange<TElement>(object target, object source)
 
         private void ActivateProperties(object controller, Type containerType, Dictionary<string, object> properties)
         {
-            var propertyHelpers = PropertyHelper.GetProperties(controller);
+            var controllerMetadata = _modelMetadataProvider.GetMetadataForType(containerType);
```

Comment:
> If you hide a property via `new` - that doesn't make it not needed by features of the base class.

Are you suggesting this code should assume the top-level property does not also initialize the hidden one? Sounds like a lot of complication, especially given issues when `base.Property` would be property initialized as-is.

Note this behaviour hasn't changed with this PR. The `properties` collection was a property name -> value mapping all along.

> I really don't think we should be trying to create a modelmetadata for a controller.

I could use the `GetMetadataForProperty()` extension method to hide the controller metadata.

Or are you suggesting we stick with `PropertyHelper` here? Why exactly?

Note `BindModelAsync()` already gets the `ModelMetadata` for the property types. But the getter and setter properties are `null` there.